CONTRIBUTING.md

```diff
@@ -56,6 +56,39 @@ So this approach has been abandoned a while ago.
 
 ### Releasing LALRPOP
 
+#### Semver policy
+
+All releases should follow rust's semantic versioning guidelines, as described
+at https://doc.rust-lang.org/cargo/reference/semver.html. Breaking changes in
+our context include changes that break previously working lalrpop grammars.
+
+If a PR introduces a breaking change, the breaking-ness should not delay the
+merge of the PR.  Instead, master will become breaking and the next release
+from master will be a breaking change release.  However, to support continued
+non-breaking releases and reduce the frequency of breaking releases, we will
+create a branch off of the last commit before the breaking change and backport
+future non-breaking changes to it.  For example, if the latest release was 0.22
+and a breaking change is merged, we will create a branch named "0.22.x" from
+the commit prior to the breaking change.  We will backport to that branch as
+non-breaking changes are merged, and possible release 0.22.1, 0.22.2 etc from
+it.
+
+Except in rare and severe circumstances, the 0.22.x branch will only be
+maintained until the release of version 0.23.  A critical bug may warrant a
+later backport to the 0.22.x series, but in general, we will stop backporting
+and focus efforts on the 0.23 series as soon as 0.23 is released.
+
+Regarding msrv bumps, the guidance in the rust community is that msrv bumps are
+not necessarily breaking, but should be bundled with breaking changes if
+possible. Our msrv strategy follows this guidance, attempting to bundle msrv
+bumps with breaking releases.  We prefer to keep the msrv at least two versions
+behind stable (eg if the latest rust release is 1.83, we prefer our msrv to be
+no more than 1.81).  In the event that there is a prolonged time between
+breaking releases, it may be desirable to bump the msrv in a non-breaking
+release.  In this case, we prefer to keep at least eight releases behind stable
+(eg if the latest rust release is 1.83, we prefer to bump the msrv in a
+non-breaking release to at most 1.75).
```

Comment:
Thinking a little bit about this part. Are the main reasons here because lalrpop want's to keep up to date with Rust features/best-practices? Another reason is what happened to us with logos, where a dependency pushed up our MSRV from under us(probably fixed in the new edition).

To clarify on eight releases, is this a magic number or approximating 1 years worth of releases(for what it's worth, the `regex` crate has msrv bumps after a year)?

Reply:
> Are the main reasons here because lalrpop want's to keep up to date with Rust features/best-practices? Another reason is what happened to us with logos, where a dependency pushed up our MSRV from under us(probably fixed in the new edition).

Pretty much, although I'll add two things:

* In addition to an existing dependency bumping the msrv, we may want to take on a new dependency with a more recent msrv (eg #981)
* The farther out of date we get, the more we have to pay attention to the version numbers in the official docs to avoid using functions that would bump the msrv.  When we do bump the msrv, we may want to remember to go forward and switch to the more modern functions where appropriate.

> To clarify on eight releases, is this a magic number or approximating 1 years worth of releases

Yes, magic number, existing basically as a strawman for discussion as much as as a real proposal.  Nine releases would approximate 1 year slightly better, but 8 is a power of 2, so I went with 8.  7+ is what would be needed to get us to 1.74 right now, which is desirable for a few open PRs.

> (for what it's worth, the regex crate has msrv bumps after a year)?

Interesting, I didn't know that.  It seems like "a year old" is probably about as old as we can reasonably hang out based on that, so reinforcement for the 8 releases threshold?